### PR TITLE
expression: implement `shiftLeft` function pushdown

### LINKED ISSUE #36733: Implement shiftLeft function push down to TiFlash
## Enhancement
support shiftLeft function push down to TiFlash.

For details: https://github.com/pingcap/tiflash/issues/5099

### PR DESCRIPTION
**Pull request by AnnieoftheStars**
### What problem does this PR solve?

Issue Number: close #36733
related issue: https://github.com/pingcap/tiflash/issues/5099

Problem Summary:

Support push down `shiftLeft` function to TiFlash

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support to pushdown leftShift to TiFlash
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -7210,6 +7210,41 @@ func TestAggWithJsonPushDownToTiFlash(t *testing.T) {
...
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int)")
+	tk.MustExec("insert into t values(2147483647, 32)")
+	tk.MustExec("insert into t values(12, 2)")
+	tk.MustExec("set @@tidb_allow_mpp=1; set @@tidb_enforce_mpp=1")
+	tk.MustExec("set @@tidb_isolation_read_engines = 'tiflash'")
+
+	// Create virtual tiflash replica info.
+	dom := domain.GetDomain(tk.Session())
```
Comment (hawkingrei) [inline planner/core/integration_test.go:7226]: use ```store, dom, clean := testkit.CreateMockStoreAndDomain(t)``` to create ```dom```.

Comment (AnnieoftheStars) [reply planner/core/integration_test.go]: done

**Review by wshwsh12 — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

**Pull request merged**